### DISCUSSION TITLE
Lazy components with listener causes render before mounted

### DISCUSSION THREAD

**Discussion by oddvalue** [livewire/livewire#7945] [7. Bugs (archived)]
### Livewire version

v3.4.4

### Laravel version

v10.*

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Set up example repo
2. Load front page
3. See error

### Repository or Wirebox URL

https://github.com/oddvalue/livewire-listener-error-example/commit/0b2173d9b4a62acb41a30fed7e9c3c11bff6c283

https://wirebox.app/b/xgrpd

### Additional information

When a listener is triggered on a lazy component it causes the component to attempt to render even if it's not yet mounted. This causes an error if there are typed properties in the component that are set on mount but over that, I wouldn't expect a component to attempt to render unless it was first mounted.

I'm not 100% sure what the fix would be or I'd have a go myself. I imagine it wants to ignore listeners until properly loaded/mounted. 

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by calebporzio** [comment]
Thank you for submitting this report - can you submit a PR with a failing test just to get the process of a fix started? Even if you can't get the tests running locally, just submitting some kind of PR will help us quite a lot. Thanks!

**Comment by oddvalue** [comment]
Hey @calebporzio I had a quick look at setting up a test but I couldn't get any of the browser tests running and I'm not sure how you'd test it without it being a browser test.

There is this wirebox example, does that help? https://wirebox.app/b/xgrpd

**Reply by joshhanley** [reply]
@oddvalue if you could write it as a browser test and just submit it (and make a note in the PR that you couldn't run the browser tests locally) that would be really helpful. Did you get any particular error when running browser tests locally? What setup are you using (OS, PHP version, web server, docker, herd, etc.)?

**Reply by oddvalue** [reply]
@joshhanley yep, no problem. I'll give it a go. 

This is the error I'm getting:

``` session not created: Chrome failed to start: exited normally.
  (session not created: DevToolsActivePort file doesn't exist)
  (The process started from chrome location /opt/google/chrome/chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
  (Driver info: chromedriver=123.0.6312.86 (9b72c47a053648d405376c5cf07999ed626728da-refs/branch-heads/6312@{#698}),platform=Linux 6.8.0-76060800daily20240311-generic x86_64)
```

On an linux system using php locally. Chrome driver is up to date and inline with the chrome version.

**Reply by oddvalue** [reply]
@joshhanley PR submitted here: #8257 

I was able to get the browser tests running by disabling the UI but my test fails with the error: 

```
javascript error: Cannot read properties of undefined (reading 'hook')
```

**Reply by danie-ramdhani** [reply]
check my comment https://github.com/livewire/livewire/pull/8257#issuecomment-2041229906

**Comment by robert-stanciu** [comment]
Hi @calebporzio, this is still happening. Is there any solution or workaround for it?
I also have lazy components with event listeners. These components get rendered without the mount method being executed.
Thanks!
